modules/configuring-ovs-log-level-permanently.adoc

```diff
@@ -0,0 +1,51 @@
+[id="configuring-ovs-log-level-permanently_{context}"]
+= Configuring the Open vSwitch log level permanently
+
+For long-term changes to the Open vSwitch (OVS) log level, you can change the log level permanently.
+
+.Prerequisites
+
+* You have access to the cluster as a user with the `cluster-admin` role.
+
+* You have installed the OpenShift CLI (`oc`).
+
+.Procedure
+
+. Create a file, such as `99-change-ovs-loglevel.yaml`, with a `MachineConfig` object like the following example:
++
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master  <.>
+  name: 99-change-ovs-loglevel
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    systemd:
+      units:
+      - dropins:
+        - contents: |
+            [Service]
+              Restart=always
+              ExecStartPre=-/bin/sh -c '/usr/bin/chown -R :$${OVS_USER_ID##*:} /var/lib/openvswitch'
+              ExecStartPre=-/bin/sh -c '/usr/bin/chown -R :$${OVS_USER_ID##*:} /etc/openvswitch'
+              ExecStartPre=-/bin/sh -c '/usr/bin/chown -R :$${OVS_USER_ID##*:} /run/openvswitch'
```

Comment:
We don't want to duplicate these lines. 

`ExecStartPre`, `ExecStartPost`, `ExecReload` are lists, each time we override we append to the list.

We only need

```ini
[Service]
ExecStartPost=-/usr/bin/ovs-appctl vlog/set syslog:debug
ExecReload=-/usr/bin/ovs-appctl vlog/set syslog:debug
```

Which generates this
```ini
sh-4.4# systemctl show ovs-vswitchd | grep 'Exec[SR].*='
ExecStartPre={ path=/bin/sh ; argv[]=/bin/sh -c /usr/bin/chown :$${OVS_USER_ID##*:} /dev/hugepages ; ignore_errors=yes ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }
ExecStartPre={ path=/usr/bin/chmod ; argv[]=/usr/bin/chmod 0775 /dev/hugepages ; ignore_errors=yes ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }
ExecStartPre={ path=/bin/sh ; argv[]=/bin/sh -c /usr/bin/chown -R :$${OVS_USER_ID##*:} /var/lib/openvswitch ; ignore_errors=yes ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }
ExecStartPre={ path=/bin/sh ; argv[]=/bin/sh -c /usr/bin/chown -R :$${OVS_USER_ID##*:} /etc/openvswitch ; ignore_errors=yes ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }
ExecStartPre={ path=/bin/sh ; argv[]=/bin/sh -c /usr/bin/chown -R :$${OVS_USER_ID##*:} /run/openvswitch ; ignore_errors=yes ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }
ExecStart={ path=/usr/share/openvswitch/scripts/ovs-ctl ; argv[]=/usr/share/openvswitch/scripts/ovs-ctl --no-ovsdb-server --no-monitor --system-id=random ${OVS_USER_OPT} start $OPTIONS ; ignore_errors=no ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }
ExecStartPost={ path=/usr/bin/ovs-appctl ; argv[]=/usr/bin/ovs-appctl vlog/set syslog:info ; ignore_errors=yes ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }
ExecStartPost={ path=/usr/bin/ovs-appctl ; argv[]=/usr/bin/ovs-appctl vlog/set syslog:debug ; ignore_errors=yes ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }
ExecReload={ path=/usr/share/openvswitch/scripts/ovs-ctl ; argv[]=/usr/share/openvswitch/scripts/ovs-ctl --no-ovsdb-server --no-monitor --system-id=random ${OVS_USER_OPT} restart $OPTIONS ; ignore_errors=no ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }
ExecReload={ path=/usr/bin/ovs-appctl ; argv[]=/usr/bin/ovs-appctl vlog/set syslog:info ; ignore_errors=yes ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }
ExecReload={ path=/usr/bin/ovs-appctl ; argv[]=/usr/bin/ovs-appctl vlog/set syslog:debug ; ignore_errors=yes ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }
ExecStop={ path=/usr/share/openvswitch/scripts/ovs-ctl ; argv[]=/usr/share/openvswitch/scripts/ovs-ctl --no-ovsdb-server stop ; ignore_errors=no ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }
```

The new `ExecStartPost=` and `ExecReload=`  commands are appended to their respective lists.

We could use an empty string to reset the list (`ExecStartPost=""`), but not for `ExecReload=` because there is the `ovs-vsctl`  command that we need to run, so we should only append commands.


Full `systemctl cat`
```ini
sh-4.4# systemctl cat ovs-vswitchd
# /usr/lib/systemd/system/ovs-vswitchd.service
[Unit]
Description=Open vSwitch Forwarding Unit
After=ovsdb-server.service network-pre.target systemd-udev-settle.service
Before=network.target network.service
Requires=ovsdb-server.service
ReloadPropagatedFrom=ovsdb-server.service
AssertPathIsReadWrite=/var/run/openvswitch/db.sock
PartOf=openvswitch.service

[Service]
Type=forking
PIDFile=/var/run/openvswitch/ovs-vswitchd.pid
Restart=on-failure
Environment=XDG_RUNTIME_DIR=/var/run/openvswitch
EnvironmentFile=/etc/openvswitch/default.conf
EnvironmentFile=-/etc/sysconfig/openvswitch
EnvironmentFile=-/run/openvswitch.useropts
LimitSTACK=2M
ExecStartPre=-/bin/sh -c '/usr/bin/chown :$${OVS_USER_ID##*:} /dev/hugepages'
ExecStartPre=-/usr/bin/chmod 0775 /dev/hugepages
ExecStart=/usr/share/openvswitch/scripts/ovs-ctl \
          --no-ovsdb-server --no-monitor --system-id=random \
          ${OVS_USER_OPT} \
          start $OPTIONS
ExecStop=/usr/share/openvswitch/scripts/ovs-ctl --no-ovsdb-server stop
ExecReload=/usr/share/openvswitch/scripts/ovs-ctl --no-ovsdb-server \
          --no-monitor --system-id=random \
          ${OVS_USER_OPT} \
          restart $OPTIONS
TimeoutSec=300

# /etc/systemd/system/ovs-vswitchd.service.d/10-ovs-vswitchd-restart.conf
[Service]
Restart=always
ExecStartPre=-/bin/sh -c '/usr/bin/chown -R :$${OVS_USER_ID##*:} /var/lib/openvswitch'
ExecStartPre=-/bin/sh -c '/usr/bin/chown -R :$${OVS_USER_ID##*:} /etc/openvswitch'
ExecStartPre=-/bin/sh -c '/usr/bin/chown -R :$${OVS_USER_ID##*:} /run/openvswitch'
ExecStartPost=-/usr/bin/ovs-appctl vlog/set syslog:info
ExecReload=-/usr/bin/ovs-appctl vlog/set syslog:info

# /etc/systemd/system/ovs-vswitchd.service.d/20-ovs-vswitchd-restart.conf
[Service]
ExecStartPost=-/usr/bin/ovs-appctl vlog/set syslog:debug
ExecReload=-/usr/bin/ovs-appctl vlog/set syslog:debug
```

Reply:
Well, jeez.  I'll revise and thank you for pointing it out to me.

Reply:
:smile:   I'm slowly trying to learn the "proper" `systemd` way to do things.